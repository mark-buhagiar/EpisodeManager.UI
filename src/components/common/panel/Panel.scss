@import '../../../variables.scss';

.panel {
    display: flex;
    position: relative;
    flex-flow: column;
    margin: 1em 0;

    .separator {
        position: absolute;
        bottom: 0;
        border-bottom: 1px solid $line-color;
        width: 75%;
    }

    .panel-title {
        position: relative;
        font-size: 2em;
        margin-bottom: 1em;
        display: flex;
        flex-flow: row;
        justify-content: space-between;
        align-items: flex-end;
    }
}
