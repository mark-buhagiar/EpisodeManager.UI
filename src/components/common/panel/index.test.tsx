import { render } from '@testing-library/react';
import React from 'react';
import Panel from './index';

describe('When the component is rendered', () => {
    it('should render the panel with all of its parts', () => {
        const panelActions = <div>Panel Actions</div>;
        const { asFragment } = render(
            <Panel title="Panel Title" panelActions={panelActions}>
                <div>Panel Content</div>
            </Panel>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
