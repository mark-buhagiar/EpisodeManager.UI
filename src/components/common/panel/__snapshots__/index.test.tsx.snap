// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the component is rendered should render the panel with all of its parts 1`] = `
<DocumentFragment>
  <div
    class="panel"
  >
    <div
      class="panel-title"
    >
      Panel Title
      <div
        class="panel-actions"
      >
        <div>
          Panel Actions
        </div>
      </div>
      <div
        class="separator"
      />
    </div>
    <div>
      Panel Content
    </div>
  </div>
</DocumentFragment>
`;
